scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -276,6 +301,242 @@ private static void unzipReportResources(InputStream source, File target) throws
+        Toml scanTomlDocumentContent;
+
+        try {
+            scanTomlDocumentContent = Toml.read(scanTomlFilePath);
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to read the configuration file with exception: " + ex.getMessage());
+        }
+        ScanTomlFile scanTomlFile = new ScanTomlFile();
+
+        List<Toml> platformTables = scanTomlDocumentContent.getTables(PLATFORM_TABLE);
+        platformTables.forEach(platformTable -> {
+            Map<String, Object> properties = platformTable.toMap();
+            String name = !(properties.get("name") instanceof String) ? null :
+                    properties.remove("name").toString();
+            String path = !(properties.get("path") instanceof String) ? null :
+                    properties.remove("path").toString();
+
+            if (name != null && !name.isEmpty() && path != null) {
+                if (!(new File(path).exists())) {
+                    try {
+                        URL url = new URL(path);
+                        path = loadRemoteJAR(targetDir, name, url, outputStream);
+                    } catch (MalformedURLException ex) {
+                        throw new RuntimeException("Failed to retrieve remote platform file with exception: "
+                                + ex.getMessage());
+                    }
+                }
+
+                if (Files.exists(Path.of(path))) {
+                    ScanTomlFile.Platform platform = new ScanTomlFile.Platform(name, path, properties);
+                    scanTomlFile.setPlatform(platform);
+                }
+            }
+        });
+
+        List<Toml> staticCodeAnalyzerTables = scanTomlDocumentContent.getTables(ANALYZER_TABLE);
+        staticCodeAnalyzerTables.forEach(staticCodeAnalyzer -> {
+            Map<String, Object> properties = staticCodeAnalyzer.toMap();
+            String org = !(properties.get("org") instanceof String) ? null :
+                    properties.get("org").toString();
+            String name = !(properties.get("name") instanceof String) ? null :
+                    properties.get("name").toString();
+
+            Pattern versionPattern = Pattern.compile(CUSTOM_RULES_COMPILER_PLUGIN_VERSION_PATTERN);
+            Object providedVersion = properties.get("version");
+            String version = (providedVersion instanceof String) &&
+                    versionPattern.matcher(providedVersion.toString()).matches() ? providedVersion.toString() : null;
+            String repository = !(properties.get("repository") instanceof String) ? null :
+                    properties.get("repository").toString();
+
+            if (org != null && !org.isEmpty() && name != null && !name.isEmpty()) {
+                ScanTomlFile.Analyzer analyzer;
+
+                if (repository != null && repository.equals(LOCAL_REPOSITORY_NAME) && version != null &&
```

Comment:
Since there's no other logic between these two if conditions, you could move the `analyzer` initializing outside the condition and combine the two conditions.